### ISSUE TITLE
This will not build against Linux 5.6.

### ISSUE THREAD

**Issue by Dale-M** [corneal64/Realtek-USB-Wireless-Adapter-Drivers#4]

**Comment by night-wielder** [comment]
Facing the same problem... Is there any working fork or something?

**Comment by arthurhscarvalho** [comment]
Same problem.

**Comment by GoliathLabs** [comment]
```diff --git a/os_dep/linux/rtw_proc.c b/os_dep/linux/rtw_proc.c
index 9ecc542..8e0f7dd 100644
--- a/os_dep/linux/rtw_proc.c
+++ b/os_dep/linux/rtw_proc.c
@@ -64,7 +64,11 @@ inline struct proc_dir_entry *rtw_proc_create_dir(const char *name, struct proc_
 }
 
 inline struct proc_dir_entry *rtw_proc_create_entry(const char *name, struct proc_dir_entry *parent, 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0))
+	const struct proc_ops *fops, void * data)
+#else
 	const struct file_operations *fops, void * data)
+#endif
 {
 	struct proc_dir_entry *entry;
 
@@ -165,6 +169,14 @@ static ssize_t rtw_drv_proc_write(struct file *file, const char __user *buffer,
 	return -EROFS;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0))
+static const struct proc_ops rtw_drv_proc_fops = {
+	.proc_open = rtw_drv_proc_open,
+	.proc_read = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
+	.proc_write = rtw_drv_proc_write,
+#else
 static const struct file_operations rtw_drv_proc_fops = {
 	.owner = THIS_MODULE,
 	.open = rtw_drv_proc_open,
@@ -172,6 +184,7 @@ static const struct file_operations rtw_drv_proc_fops = {
 	.llseek = seq_lseek,
 	.release = single_release,
 	.write = rtw_drv_proc_write,
+#endif
 };
 
 int rtw_drv_proc_init(void)
@@ -532,6 +545,14 @@ static ssize_t rtw_adapter_proc_write(struct file *file, const char __user *buff
 	return -EROFS;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0))
+static const struct proc_ops rtw_adapter_proc_fops = {
+	.proc_open = rtw_adapter_proc_open,
+	.proc_read = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
+	.proc_write = rtw_adapter_proc_write,
+#else
 static const struct file_operations rtw_adapter_proc_fops = {
 	.owner = THIS_MODULE,
 	.open = rtw_adapter_proc_open,
@@ -539,6 +560,7 @@ static const struct file_operations rtw_adapter_proc_fops = {
 	.llseek = seq_lseek,
 	.release = single_release,
 	.write = rtw_adapter_proc_write,
+#endif
 };
 
 int proc_get_odm_dbg_comp(struct seq_file *m, void *v)
@@ -710,6 +732,14 @@ static ssize_t rtw_odm_proc_write(struct file *file, const char __user *buffer,
 	return -EROFS;
 }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0))
+static const struct proc_ops rtw_odm_proc_fops = {
+	.proc_open = rtw_odm_proc_open,
+	.proc_read = seq_read,
+	.proc_lseek = seq_lseek,
+	.proc_release = single_release,
+	.proc_write = rtw_odm_proc_write,
+#else
 static const struct file_operations rtw_odm_proc_fops = {
 	.owner = THIS_MODULE,
 	.open = rtw_odm_proc_open,
@@ -717,6 +747,7 @@ static const struct file_operations rtw_odm_proc_fops = {
 	.llseek = seq_lseek,
 	.release = single_release,
 	.write = rtw_odm_proc_write,
+#endif
 };
 
 struct proc_dir_entry *rtw_odm_proc_init(struct net_device *dev)
```

A patch file :)

**Comment by duttasukanta** [comment]
Ooooo, Thanks a lot. It Works finally with Kernel 5.6. I used Kali Linux 2020.2.Initially I got error with the command
$ sudo make install.
[error in line 428(Makefile (install -p -m 644 $(MODULE_NAME).ko  $(MODDESTDIR)  ..]
It was missing the $(module_name).ko
Later I manually changed the $(module_name).ko with rtl8188fu.ko in the Makefile and it installed successfully.
Of course I have applied the patch above mentioned.
Thank you.
